docs_src/design/adr/security/tbd-authentication-e2ee.md

```diff
@@ -0,0 +1,414 @@
+However, to be fully functional,
+and adopter must also run an OpenZiti controller,
+one or more OpenZiti router components,
+and zero or more OpenZiti tunnelers to support legacy applications.
+
+In a zero-trust architecture, an API gateway is no longer required
+for remote access to EdgeX microservice API's.
+If an API gateway is still desired,
+it is still possible to run the EdgeX 3.0 API gateway
+in conjunction with an OpenZiti tunneler to onboard
+API gateway traffic onto the zero-trust network on the backend.
+
+The new TO-BE architecture is diagrammed in the following figure:
+
+![TO-BE Architecture](tbd-to-be.jpg)
+
+This diagram conveys a lot of information in little space.
+A little exposition on what this diagram is intending to show:
+
+The yellow boxes,
+showing the OpenZiti Controller, OpenZiti Router,
+and OpenZiti Tunneler are the new OpenZiti components.
+Two of these components, the controller and the router,
+are in a dashed box, denoting that these components
+have open ports on the network underlay.
+The controller and the router could be Internet-accessible,
+on a local docker network, or on the host network.
+To permit off-host access to the zero-trust network,
+either the OpenZiti router must be exposed externally on the host,
+or there must be a connection to a peer router that
+is itself accessible off-host.
+"Zitified" clients can simply connect to an OpenZiti edge router
+and join the zero-trust network.
+Non-ziti-aware components such as browsers or command-line clients
+must run on a host with an OpenZiti tunneler
+that works similar to an SSH port-forwarder to
+forward connections onto the zero-trust network.
+There is some security risk in doing this,
+as the OpenZiti Tunneler can't tell the difference between
+a non-zitified client and malware running on the host.
+
+The green boxes,
+showing Vault, Redis, and Consul,
+are existing third-party non-zitified components
+that are already used by EdgeX.
+These components have independent authentication mechanisms.
+They are safe to use for kernel-mediated same-host access,
+but not safe to access over the network without
+additional network security.
+The OpenZiti Router also has an outbound tunneler built-in,
+allowing traffic to traverse the zero-trust network encrypted,
+and exit unencrypted to communicate with services
+listening on the docker network (zero-trust network access, ZTNA),
+or listening on the host network (zero-trust host access, ZTHA).
+These connections are indicated by the blue-violet arrows.
+Note that Vault also has a thick dashed border,
+indicating that it also must have an exposed listening port.
+This is because a Vault-based identity claim is required
+to join the zero-trust network in the normal case,
+and thus Vault must be available as a prerequisite
+to the zero-trust network being available.
+In EdgeX 3.0,
+Vault is exposed behind a TLS-enabled NGINX API gateway (not shown),
```

Comment:
From statement above, the`API gateway` is no longer needed. How does this impact this statement?

Reply:
Good point.  I was trying to say that Vault still needs to be TLS-enabled regardless.  An API gateway is not the only option.  I have rewritten as such:

As Vault security is paramount,
secure network communication with Vault must be possible,
even before the zero-trust network is brought up.
Possible options, at the option of the adopter, include:
* Vault is itself TLS-enabled
* Vault resides behind TLS-enabled API gateway
* Vault resides behind some kind of secure tunnel (e.g `stunnel`)
* Vault is accessible via an encrypted VPN